File: php/lib/Model/ConvertAmounts200Response.php
<?php
/**
 * ConvertAmounts200Response
 *
 * PHP version 7.4
 *
 * @category Class
 * @package  OpenAPI\Client
 * @author   OpenAPI Generator team
 * @link     https://openapi-generator.tech
 */

/**
 * spoonacular API
 *
 * The spoonacular Nutrition, Recipe, and Food API allows you to access over thousands of recipes, thousands of ingredients, 800,000 food products, over 100,000 menu items, and restaurants. Our food ontology and semantic recipe search engine makes it possible to search for recipes using natural language queries, such as \"gluten free brownies without sugar\" or \"low fat vegan cupcakes.\" You can automatically calculate the nutritional information for any recipe, analyze recipe costs, visualize ingredient lists, find recipes for what's in your fridge, find recipes based on special diets, nutritional requirements, or favorite ingredients, classify recipes into types and cuisines, convert ingredient amounts, or even compute an entire meal plan. With our powerful API, you can create many kinds of food and especially nutrition apps.  Special diets/dietary requirements currently available include: vegan, vegetarian, pescetarian, gluten free, grain free, dairy free, high protein, whole 30, low sodium, low carb, Paleo, ketogenic, FODMAP, and Primal.
 *
 * The version of the OpenAPI document: 1.1
 * Contact: mail@spoonacular.com
 * Generated by: https://openapi-generator.tech
 * OpenAPI Generator version: 7.3.0
 */

/**
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

namespace OpenAPI\Client\Model;

use \ArrayAccess;
use \OpenAPI\Client\ObjectSerializer;

/**
 * ConvertAmounts200Response Class Doc Comment
 *
 * @category Class
 * @description 
 * @package  OpenAPI\Client
 * @author   OpenAPI Generator team
 * @link     https://openapi-generator.tech
 * @implements \ArrayAccess<string, mixed>
 */
class ConvertAmounts200Response implements ModelInterface, ArrayAccess, \JsonSerializable
{
    public const DISCRIMINATOR = null;

    /**
      * The original name of the model.
      *
      * @var string
      */
    protected static $openAPIModelName = 'convertAmounts_200_response';

    /**
      * Array of property to type mappings. Used for (de)serialization
      *
      * @var string[]
      */
    protected static $openAPITypes = [
        'source_amount' => 'float',
        'source_unit' => 'string',
        'target_amount' => 'float',
        'target_unit' => 'string',
        'answer' => 'string'
    ];

    /**
      * Array of property to format mappings. Used for (de)serialization
      *
      * @var string[]
      * @phpstan-var array<string, string|null>
      * @psalm-var array<string, string|null>
      */
    protected static $openAPIFormats = [
        'source_amount' => null,
        'source_unit' => null,
        'target_amount' => null,
        'target_unit' => null,
        'answer' => null
    ];

    /**
      * Array of nullable properties. Used for (de)serialization
      *
      * @var boolean[]
      */
    protected static array $openAPINullables = [
        'source_amount' => false,
        'source_unit' => false,
        'target_amount' => false,
        'target_unit' => false,
        'answer' => false
    ];

    /**
      * If a nullable field gets set to null, insert it here
      *
      * @var boolean[]
      */
    protected array $openAPINullablesSetToNull = [];

    /**
     * Array of property to type mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function openAPITypes()
    {
        return self::$openAPITypes;
    }

    /**
     * Array of property to format mappings. Used for (de)serialization
     *
     * @return array
     */
    public static function openAPIFormats()
    {
        return self::$openAPIFormats;
    }

    /**
     * Array of nullable properties
     *
     * @return array
     */
    protected static function openAPINullables(): array
    {
        return self::$openAPINullables;
    }

    /**
     * Array of nullable field names deliberately set to null
     *
     * @return boolean[]
     */
    private function getOpenAPINullablesSetToNull(): array
    {
        return $this->openAPINullablesSetToNull;
    }

    /**
     * Setter - Array of nullable field names deliberately set to null
     *
     * @param boolean[] $openAPINullablesSetToNull
     */
    private function setOpenAPINullablesSetToNull(array $openAPINullablesSetToNull): void
    {
        $this->openAPINullablesSetToNull = $openAPINullablesSetToNull;
    }

    /**
     * Checks if a property is nullable
     *
     * @param string $property
     * @return bool
     */
    public static function isNullable(string $property): bool
    {
        return self::openAPINullables()[$property] ?? false;
    }

    /**
     * Checks if a nullable property is set to null.
     *
     * @param string $property
     * @return bool
     */
    public function isNullableSetToNull(string $property): bool
    {
        return in_array($property, $this->getOpenAPINullablesSetToNull(), true);
    }

    /**
     * Array of attributes where the key is the local name,
     * and the value is the original name
     *
     * @var string[]
     */
    protected static $attributeMap = [
        'source_amount' => 'sourceAmount',
        'source_unit' => 'sourceUnit',
        'target_amount' => 'targetAmount',
        'target_unit' => 'targetUnit',
        'answer' => 'answer'
    ];

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @var string[]
     */
    protected static $setters = [
        'source_amount' => 'setSourceAmount',
        'source_unit' => 'setSourceUnit',
        'target_amount' => 'setTargetAmount',
        'target_unit' => 'setTargetUnit',
        'answer' => 'setAnswer'
    ];

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @var string[]
     */
    protected static $getters = [
        'source_amount' => 'getSourceAmount',
        'source_unit' => 'getSourceUnit',
        'target_amount' => 'getTargetAmount',
        'target_unit' => 'getTargetUnit',
        'answer' => 'getAnswer'
    ];

    /**
     * Array of attributes where the key is the local name,
     * and the value is the original name
     *
     * @return array
     */
    public static function attributeMap()
    {
        return self::$attributeMap;
    }

    /**
     * Array of attributes to setter functions (for deserialization of responses)
     *
     * @return array
     */
    public static function setters()
    {
        return self::$setters;
    }

    /**
     * Array of attributes to getter functions (for serialization of requests)
     *
     * @return array
     */
    public static function getters()
    {
        return self::$getters;
    }

    /**
     * The original name of the model.
     *
     * @return string
     */
    public function getModelName()
    {
        return self::$openAPIModelName;
    }


    /**
     * Associative array for storing property values
     *
     * @var mixed[]
     */
    protected $container = [];

    /**
     * Constructor
     *
     * @param mixed[] $data Associated array of property values
     *                      initializing the model
     */
    public function __construct(array $data = null)
    {
        $this->setIfExists('source_amount', $data ?? [], null);
        $this->setIfExists('source_unit', $data ?? [], null);
        $this->setIfExists('target_amount', $data ?? [], null);
        $this->setIfExists('target_unit', $data ?? [], null);
        $this->setIfExists('answer', $data ?? [], null);
    }

    /**
    * Sets $this->container[$variableName] to the given data or to the given default Value; if $variableName
    * is nullable and its value is set to null in the $fields array, then mark it as "set to null" in the
    * $this->openAPINullablesSetToNull array
    *
    * @param string $variableName
    * @param array  $fields
    * @param mixed  $defaultValue
    */
    private function setIfExists(string $variableName, array $fields, $defaultValue): void
    {
        if (self::isNullable($variableName) && array_key_exists($variableName, $fields) && is_null($fields[$variableName])) {
            $this->openAPINullablesSetToNull[] = $variableName;
        }

        $this->container[$variableName] = $fields[$variableName] ?? $defaultValue;
    }

    /**
     * Show all the invalid properties with reasons.
     *
     * @return array invalid properties with reasons
     */
    public function listInvalidProperties()
    {
        $invalidProperties = [];

        if ($this->container['source_amount'] === null) {
            $invalidProperties[] = "'source_amount' can't be null";
        }
        if ($this->container['source_unit'] === null) {
            $invalidProperties[] = "'source_unit' can't be null";
        }
        if ((mb_strlen($this->container['source_unit']) < 1)) {
            $invalidProperties[] = "invalid value for 'source_unit', the character length must be bigger than or equal to 1.";
        }

        if ($this->container['target_amount'] === null) {
            $invalidProperties[] = "'target_amount' can't be null";
        }
        if ($this->container['target_unit'] === null) {
            $invalidProperties[] = "'target_unit' can't be null";
        }
        if ((mb_strlen($this->container['target_unit']) < 1)) {
            $invalidProperties[] = "invalid value for 'target_unit', the character length must be bigger than or equal to 1.";
        }

        if ($this->container['answer'] === null) {
            $invalidProperties[] = "'answer' can't be null";
        }
        if ((mb_strlen($this->container['answer']) < 1)) {
            $invalidProperties[] = "invalid value for 'answer', the character length must be bigger than or equal to 1.";
        }

        return $invalidProperties;
    }

    /**
     * Validate all the properties in the model
     * return true if all passed
     *
     * @return bool True if all properties are valid
     */
    public function valid()
    {
        return count($this->listInvalidProperties()) === 0;
    }


    /**
     * Gets source_amount
     *
     * @return float
     */
    public function getSourceAmount()
    {
        return $this->container['source_amount'];
    }

    /**
     * Sets source_amount
     *
     * @param float $source_amount source_amount
     *
     * @return self
     */
    public function setSourceAmount($source_amount)
    {
        if (is_null($source_amount)) {
            throw new \InvalidArgumentException('non-nullable source_amount cannot be null');
        }
        $this->container['source_amount'] = $source_amount;

        return $this;
    }

    /**
     * Gets source_unit
     *
     * @return string
     */
    public function getSourceUnit()
    {
        return $this->container['source_unit'];
    }

    /**
     * Sets source_unit
     *
     * @param string $source_unit source_unit
     *
     * @return self
     */
    public function setSourceUnit($source_unit)
    {
        if (is_null($source_unit)) {
            throw new \InvalidArgumentException('non-nullable source_unit cannot be null');
        }

        if ((mb_strlen($source_unit) < 1)) {
            throw new \InvalidArgumentException('invalid length for $source_unit when calling ConvertAmounts200Response., must be bigger than or equal to 1.');
        }

        $this->container['source_unit'] = $source_unit;

        return $this;
    }

    /**
     * Gets target_amount
     *
     * @return float
     */
    public function getTargetAmount()
    {
        return $this->container['target_amount'];
    }

    /**
     * Sets target_amount
     *
     * @param float $target_amount target_amount
     *
     * @return self
     */
    public function setTargetAmount($target_amount)
    {
        if (is_null($target_amount)) {
            throw new \InvalidArgumentException('non-nullable target_amount cannot be null');
        }
        $this->container['target_amount'] = $target_amount;

        return $this;
    }

    /**
     * Gets target_unit
     *
     * @return string
     */
    public function getTargetUnit()
    {
        return $this->container['target_unit'];
    }

    /**
     * Sets target_unit
     *
     * @param string $target_unit target_unit
     *
     * @return self
     */
    public function setTargetUnit($target_unit)
    {
        if (is_null($target_unit)) {
            throw new \InvalidArgumentException('non-nullable target_unit cannot be null');
        }

        if ((mb_strlen($target_unit) < 1)) {
            throw new \InvalidArgumentException('invalid length for $target_unit when calling ConvertAmounts200Response., must be bigger than or equal to 1.');
        }

        $this->container['target_unit'] = $target_unit;

        return $this;
    }

    /**
     * Gets answer
     *
     * @return string
     */
    public function getAnswer()
    {
        return $this->container['answer'];
    }

    /**
     * Sets answer
     *
     * @param string $answer answer
     *
     * @return self
     */
    public function setAnswer($answer)
    {
        if (is_null($answer)) {
            throw new \InvalidArgumentException('non-nullable answer cannot be null');
        }

        if ((mb_strlen($answer) < 1)) {
            throw new \InvalidArgumentException('invalid length for $answer when calling ConvertAmounts200Response., must be bigger than or equal to 1.');
        }

        $this->container['answer'] = $answer;

        return $this;
    }
    /**
     * Returns true if offset exists. False otherwise.
     *
     * @param integer $offset Offset
     *
     * @return boolean
     */
    public function offsetExists($offset): bool
    {
        return isset($this->container[$offset]);
    }

    /**
     * Gets offset.
     *
     * @param integer $offset Offset
     *
     * @return mixed|null
     */
    #[\ReturnTypeWillChange]
    public function offsetGet($offset)
    {
        return $this->container[$offset] ?? null;
    }

    /**
     * Sets value based on offset.
     *
     * @param int|null $offset Offset
     * @param mixed    $value  Value to be set
     *
     * @return void
     */
    public function offsetSet($offset, $value): void
    {
        if (is_null($offset)) {
            $this->container[] = $value;
        } else {
            $this->container[$offset] = $value;
        }
    }

    /**
     * Unsets offset.
     *
     * @param integer $offset Offset
     *
     * @return void
     */
    public function offsetUnset($offset): void
    {
        unset($this->container[$offset]);
    }

    /**
     * Serializes the object to a value that can be serialized natively by json_encode().
     * @link https://www.php.net/manual/en/jsonserializable.jsonserialize.php
     *
     * @return mixed Returns data which can be serialized by json_encode(), which is a value
     * of any type other than a resource.
     */
    #[\ReturnTypeWillChange]
    public function jsonSerialize()
    {
       return ObjectSerializer::sanitizeForSerialization($this);
    }

    /**
     * Gets the string presentation of the object
     *
     * @return string
     */
    public function __toString()
    {
        return json_encode(
            ObjectSerializer::sanitizeForSerialization($this),
            JSON_PRETTY_PRINT
        );
    }

    /**
     * Gets a header-safe presentation of the object
     *
     * @return string
     */
    public function toHeaderValue()
    {
        return json_encode(ObjectSerializer::sanitizeForSerialization($this));
    }
}

File: java/src/main/java/com/spoonacular/client/model/ConvertAmounts200Response.java
/*
 * spoonacular API
 * The spoonacular Nutrition, Recipe, and Food API allows you to access over thousands of recipes, thousands of ingredients, 800,000 food products, over 100,000 menu items, and restaurants. Our food ontology and semantic recipe search engine makes it possible to search for recipes using natural language queries, such as \"gluten free brownies without sugar\" or \"low fat vegan cupcakes.\" You can automatically calculate the nutritional information for any recipe, analyze recipe costs, visualize ingredient lists, find recipes for what's in your fridge, find recipes based on special diets, nutritional requirements, or favorite ingredients, classify recipes into types and cuisines, convert ingredient amounts, or even compute an entire meal plan. With our powerful API, you can create many kinds of food and especially nutrition apps.  Special diets/dietary requirements currently available include: vegan, vegetarian, pescetarian, gluten free, grain free, dairy free, high protein, whole 30, low sodium, low carb, Paleo, ketogenic, FODMAP, and Primal.
 *
 * The version of the OpenAPI document: 1.1
 * Contact: mail@spoonacular.com
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


package com.spoonacular.client.model;

import java.util.Objects;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;
import java.math.BigDecimal;
import java.util.Arrays;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.JsonArray;
import com.google.gson.JsonDeserializationContext;
import com.google.gson.JsonDeserializer;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.google.gson.JsonParseException;
import com.google.gson.TypeAdapterFactory;
import com.google.gson.reflect.TypeToken;
import com.google.gson.TypeAdapter;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import java.io.IOException;

import java.lang.reflect.Type;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;

import com.spoonacular.client.JSON;

/**
 * 
 */
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.JavaClientCodegen", comments = "Generator version: 7.7.0-SNAPSHOT")
public class ConvertAmounts200Response {
  public static final String SERIALIZED_NAME_SOURCE_AMOUNT = "sourceAmount";
  @SerializedName(SERIALIZED_NAME_SOURCE_AMOUNT)
  private BigDecimal sourceAmount;

  public static final String SERIALIZED_NAME_SOURCE_UNIT = "sourceUnit";
  @SerializedName(SERIALIZED_NAME_SOURCE_UNIT)
  private String sourceUnit;

  public static final String SERIALIZED_NAME_TARGET_AMOUNT = "targetAmount";
  @SerializedName(SERIALIZED_NAME_TARGET_AMOUNT)
  private BigDecimal targetAmount;

  public static final String SERIALIZED_NAME_TARGET_UNIT = "targetUnit";
  @SerializedName(SERIALIZED_NAME_TARGET_UNIT)
  private String targetUnit;

  public static final String SERIALIZED_NAME_ANSWER = "answer";
  @SerializedName(SERIALIZED_NAME_ANSWER)
  private String answer;

  public ConvertAmounts200Response() {
  }

  public ConvertAmounts200Response sourceAmount(BigDecimal sourceAmount) {
    this.sourceAmount = sourceAmount;
    return this;
  }

  /**
   * Get sourceAmount
   * @return sourceAmount
   */
  @javax.annotation.Nonnull
  public BigDecimal getSourceAmount() {
    return sourceAmount;
  }

  public void setSourceAmount(BigDecimal sourceAmount) {
    this.sourceAmount = sourceAmount;
  }


  public ConvertAmounts200Response sourceUnit(String sourceUnit) {
    this.sourceUnit = sourceUnit;
    return this;
  }

  /**
   * Get sourceUnit
   * @return sourceUnit
   */
  @javax.annotation.Nonnull
  public String getSourceUnit() {
    return sourceUnit;
  }

  public void setSourceUnit(String sourceUnit) {
    this.sourceUnit = sourceUnit;
  }


  public ConvertAmounts200Response targetAmount(BigDecimal targetAmount) {
    this.targetAmount = targetAmount;
    return this;
  }

  /**
   * Get targetAmount
   * @return targetAmount
   */
  @javax.annotation.Nonnull
  public BigDecimal getTargetAmount() {
    return targetAmount;
  }

  public void setTargetAmount(BigDecimal targetAmount) {
    this.targetAmount = targetAmount;
  }


  public ConvertAmounts200Response targetUnit(String targetUnit) {
    this.targetUnit = targetUnit;
    return this;
  }

  /**
   * Get targetUnit
   * @return targetUnit
   */
  @javax.annotation.Nonnull
  public String getTargetUnit() {
    return targetUnit;
  }

  public void setTargetUnit(String targetUnit) {
    this.targetUnit = targetUnit;
  }


  public ConvertAmounts200Response answer(String answer) {
    this.answer = answer;
    return this;
  }

  /**
   * Get answer
   * @return answer
   */
  @javax.annotation.Nonnull
  public String getAnswer() {
    return answer;
  }

  public void setAnswer(String answer) {
    this.answer = answer;
  }



  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    ConvertAmounts200Response convertAmounts200Response = (ConvertAmounts200Response) o;
    return Objects.equals(this.sourceAmount, convertAmounts200Response.sourceAmount) &&
        Objects.equals(this.sourceUnit, convertAmounts200Response.sourceUnit) &&
        Objects.equals(this.targetAmount, convertAmounts200Response.targetAmount) &&
        Objects.equals(this.targetUnit, convertAmounts200Response.targetUnit) &&
        Objects.equals(this.answer, convertAmounts200Response.answer);
  }

  @Override
  public int hashCode() {
    return Objects.hash(sourceAmount, sourceUnit, targetAmount, targetUnit, answer);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class ConvertAmounts200Response {\n");
    sb.append("    sourceAmount: ").append(toIndentedString(sourceAmount)).append("\n");
    sb.append("    sourceUnit: ").append(toIndentedString(sourceUnit)).append("\n");
    sb.append("    targetAmount: ").append(toIndentedString(targetAmount)).append("\n");
    sb.append("    targetUnit: ").append(toIndentedString(targetUnit)).append("\n");
    sb.append("    answer: ").append(toIndentedString(answer)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }


  public static HashSet<String> openapiFields;
  public static HashSet<String> openapiRequiredFields;

  static {
    // a set of all properties/fields (JSON key names)
    openapiFields = new HashSet<String>();
    openapiFields.add("sourceAmount");
    openapiFields.add("sourceUnit");
    openapiFields.add("targetAmount");
    openapiFields.add("targetUnit");
    openapiFields.add("answer");

    // a set of required properties/fields (JSON key names)
    openapiRequiredFields = new HashSet<String>();
    openapiRequiredFields.add("sourceAmount");
    openapiRequiredFields.add("sourceUnit");
    openapiRequiredFields.add("targetAmount");
    openapiRequiredFields.add("targetUnit");
    openapiRequiredFields.add("answer");
  }

  /**
   * Validates the JSON Element and throws an exception if issues found
   *
   * @param jsonElement JSON Element
   * @throws IOException if the JSON Element is invalid with respect to ConvertAmounts200Response
   */
  public static void validateJsonElement(JsonElement jsonElement) throws IOException {
      if (jsonElement == null) {
        if (!ConvertAmounts200Response.openapiRequiredFields.isEmpty()) { // has required fields but JSON element is null
          throw new IllegalArgumentException(String.format("The required field(s) %s in ConvertAmounts200Response is not found in the empty JSON string", ConvertAmounts200Response.openapiRequiredFields.toString()));
        }
      }

      Set<Map.Entry<String, JsonElement>> entries = jsonElement.getAsJsonObject().entrySet();
      // check to see if the JSON string contains additional fields
      for (Map.Entry<String, JsonElement> entry : entries) {
        if (!ConvertAmounts200Response.openapiFields.contains(entry.getKey())) {
          throw new IllegalArgumentException(String.format("The field `%s` in the JSON string is not defined in the `ConvertAmounts200Response` properties. JSON: %s", entry.getKey(), jsonElement.toString()));
        }
      }

      // check to make sure all required properties/fields are present in the JSON string
      for (String requiredField : ConvertAmounts200Response.openapiRequiredFields) {
        if (jsonElement.getAsJsonObject().get(requiredField) == null) {
          throw new IllegalArgumentException(String.format("The required field `%s` is not found in the JSON string: %s", requiredField, jsonElement.toString()));
        }
      }
        JsonObject jsonObj = jsonElement.getAsJsonObject();
      if (!jsonObj.get("sourceUnit").isJsonPrimitive()) {
        throw new IllegalArgumentException(String.format("Expected the field `sourceUnit` to be a primitive type in the JSON string but got `%s`", jsonObj.get("sourceUnit").toString()));
      }
      if (!jsonObj.get("targetUnit").isJsonPrimitive()) {
        throw new IllegalArgumentException(String.format("Expected the field `targetUnit` to be a primitive type in the JSON string but got `%s`", jsonObj.get("targetUnit").toString()));
      }
      if (!jsonObj.get("answer").isJsonPrimitive()) {
        throw new IllegalArgumentException(String.format("Expected the field `answer` to be a primitive type in the JSON string but got `%s`", jsonObj.get("answer").toString()));
      }
  }

  public static class CustomTypeAdapterFactory implements TypeAdapterFactory {
    @SuppressWarnings("unchecked")
    @Override
    public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> type) {
       if (!ConvertAmounts200Response.class.isAssignableFrom(type.getRawType())) {
         return null; // this class only serializes 'ConvertAmounts200Response' and its subtypes
       }
       final TypeAdapter<JsonElement> elementAdapter = gson.getAdapter(JsonElement.class);
       final TypeAdapter<ConvertAmounts200Response> thisAdapter
                        = gson.getDelegateAdapter(this, TypeToken.get(ConvertAmounts200Response.class));

       return (TypeAdapter<T>) new TypeAdapter<ConvertAmounts200Response>() {
           @Override
           public void write(JsonWriter out, ConvertAmounts200Response value) throws IOException {
             JsonObject obj = thisAdapter.toJsonTree(value).getAsJsonObject();
             elementAdapter.write(out, obj);
           }

           @Override
           public ConvertAmounts200Response read(JsonReader in) throws IOException {
             JsonElement jsonElement = elementAdapter.read(in);
             validateJsonElement(jsonElement);
             return thisAdapter.fromJsonTree(jsonElement);
           }

       }.nullSafe();
    }
  }

  /**
   * Create an instance of ConvertAmounts200Response given an JSON string
   *
   * @param jsonString JSON string
   * @return An instance of ConvertAmounts200Response
   * @throws IOException if the JSON string is invalid with respect to ConvertAmounts200Response
   */
  public static ConvertAmounts200Response fromJson(String jsonString) throws IOException {
    return JSON.getGson().fromJson(jsonString, ConvertAmounts200Response.class);
  }

  /**
   * Convert an instance of ConvertAmounts200Response to an JSON string
   *
   * @return JSON string
   */
  public String toJson() {
    return JSON.getGson().toJson(this);
  }
}

